src/BCReader.jl

```diff
@@ -0,0 +1,310 @@
+#=
+    BCReader
+
+This module coordinates reading of boundary conditions from NetCDF files,
+as well as regridding calls and temporal interpolations from
+monthly to daily intervals.
+=#
+module BCReader
+
+using ClimaCoupler: ClimaCoupler, Utilities, Regridder, CallbackManager
```

Comment:
```suggestion
using ..: Utilities, Regridder, CallbackManager
```
This is probably safer (`using ClimaCoupler` can result in using multiple versions of the package, and can cause very confusing errors)